receiver/fluentforwardreceiver/server.go

```diff
@@ -0,0 +1,116 @@
+package fluentforwardreceiver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync/atomic"
+
+	"github.com/tinylib/msgp/msgp"
+	"go.opentelemetry.io/collector/receiver/fluentforwardreceiver/protocol"
+	"go.uber.org/zap"
+)
+
+// The size of the read buffer.  This is also effectively the max size of any
+// single msgpack array sent from the client.
+const readBufferSize = 10 * 1024
```

Comment:
Sorry, comment was wrong, will update it.  That is just the initial buffer size, not the max possible.  If the decoder ever needed to read more than this size in a single Read it would resize the buffer bigger but that is unlikely.